consensus/src/detail/error.rs

```diff
@@ -81,7 +90,15 @@ pub enum BlockError {
     SerializationInvariantError(Id<Block>),
     #[error("Unexpected numeric type conversion error `{0:?}`")]
     InternalNumTypeConversionError(Id<Block>),
-    // To be expanded
+    #[error("Conversion failed: `{0:?}`")]
+    Conversion(String),
+    #[error("No PoW data for block")]
+    NoPowData,
+    #[error("Block consensus data is {block_consensus_data:?}, current config is {chain_consensus_status:?}")]
+    ConsensusTypeMismatch {
+        block_consensus_data: ConsensusData,
```

Comment:
Consensus data is relatively large. You can't print it in an error. You can make a string value out of it that mentions the type.

Reply:
Done.